lesson_02_homework.rb

```diff
@@ -1,46 +1,91 @@
-# Code Reading
 
+#def=Method, assigning to full_name 3 different variables -->
 def full_name(first_name, last_name, title)
-  # Example comment line
+  <!-- # Example comment line
 
+  #name assigned as nil -->
   name = nil
-
+  
+  <!-- #setting the first scenario/condition when running code,if it comes true then puts the tittle first then space, first name, space then last name. -->
   if title && first_name && last_name
     name = title + " " + first_name + " " + last_name
-  elsif title && last_name
+ # <!-- this send case is only when given title and last name -->
+ #  elsif title && last_name
     name = title + " " + last_name
+    # <!-- the third condition with first and last name -->
   elsif first_name && last_name
     name = first_name + " " + last_name
+    # <!-- only the first name -->
   elsif first_name
     name = first_name
+    # <!-- otherwhise just comes as this line of sentence telling that the name wasnt writen properly -->
   else
     raise "Oh no, that doesn't look like a name"
   end
-
+  # <!-- prints the correct statement that matched the code above -->
   return name
+  # <!-- closing the def method -->
 end
 
+
 # Fix Broken Code
 
+
 def full_name(f, l)
```

Comment:
🔨 @loriksm  the parameter names are too general f and l do not describe the value that is expected to be stored in the variable